app/src/main/java/eg/gov/iti/jets/newsapp/newsdetails/presentation/model/NewsDetailsModel.kt

```diff
@@ -0,0 +1,43 @@
+package eg.gov.iti.jets.newsapp.newsdetails.presentation.model
+
+import android.os.Parcel
+import android.os.Parcelable
+
+data class NewsDetailsModel(
+    val title: String?,
+    val authorName: String?,
+    val description: String?,
+    val imageUrl: String?,
+    var isFavorite:Boolean
+): Parcelable {
+    constructor(parcel: Parcel) : this(
```

Comment:
you can use [this](https://developer.android.com/kotlin/parcelize) to achive the same result with @Parcelize annotiotin